scripts/Auto-install-macos.sh

```diff
@@ -0,0 +1,149 @@
+  echo "Name can be anything, authorization type must be 'OAuth 2 authorization without a callback URL'"
+  echo "Press submit to get your client ID"
+  echo "NOTE: Port formwarding must be enabled on your server to send images"
+  vared -p "Imgur client ID (): " -c imgurId
+fi
+
+#put together the config file
+
+configFileContents=""
+configFileContents+="$(cat <<EOF
+{
+    "Jellyfin": {
+        "URL": "${jellyfinurl}",
+        "API_KEY": "${jellyfinkey}",
+        "USERNAME": "${jellyfinuser}"
+EOF
+)"
+if [[ ${#responses} -ne 0 ]]; then # only add this line if there are items in the blocklist
+  configFileContents+="$(cat <<EOF
+,
+        "BLOCKLIST": ${blocklistString}
+EOF
+)"
+fi
+configFileContents+="$(cat <<EOF
+
+    },
+    "Discord": {
+        "APPLICATION_ID": "${discordAppId}"
+    },
+    "Imgur": {
+        "CLIENT_ID": "${imgurId}"
+    },
+    "Images": {
+        "ENABLE_IMAGES": ${configImgurEnabled},
+        "IMGUR_IMAGES": ${configImgurEnabled}
+    }
+}
+EOF
+)"
+
+# Preview config file and have user confirm before saving it
+echo "Config file preview:"
+echo $configFileContents
+vared -p "Save it and proceed? (y/n) " -c proceed
+if [[ $proceed == [Nn]* ]]; then
+  exit
+fi
+
+#Save config to file
+if [ ! -d ~/.config/jellyfin-rpc ]; then
+  mkdir ~/.config/jellyfin-rpc
+fi
+echo $configFileContents > ~/.config/jellyfin-rpc/main.json
+
+# Prompt user to install Jellyfin RPC
+vared -p "Download latest release? (y/n) " -c downloadlatest
+if [[ $downloadlatest == [Yy]* ]]; then
+  # download file to binary directory and give execution permissions
+  curl -o /usr/local/bin/jellyfin-rpc -L https://github.com/Radiicall/jellyfin-rpc/releases/latest/download/jellyfin-rpc-x86_64-darwin
+  chmod +x /usr/local/bin/jellyfin-rpc
+
+  # Prompt user to enabled Jellyfin RPC running at login
+  vared -p "Set Jellyfin-RPC to run at login? (y/n) " -c runAtLogin
+  if [[ $runAtLogin == [Yy]* ]]; then
+    # Create launch agent file, give it proper permissions, then load it.
+    cat >> ~/Library/LaunchAgents/jellyfinrpc.local.plist<< EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>Label</key>
+    <string>Jellyfin-RPC</string>
+    <key>Program</key>
+    <string>/usr/local/bin/jellyfin-rpc</string>
+    <key>RunAtLoad</key>
+    <true/>
+    <key>StandardErrorPath</key>
+    <string>/tmp/jellyfinrpc.local.stderr.txt</string>
+    <key>StandardOutPath</key>
+    <string>/tmp/jellyfinrpc.local.stdout.txt</string>
+</dict>
+</plist>
+EOF
+    chmod 644 ~/Library/LaunchAgents/jellyfinrpc.local.plist
+    launchctl unload ~/Library/LaunchAgents/jellyfinrpc.local.plist #It seems that the OS tries to load the launch agent on creation, so it needs unloaded first.
+    launchctl load ~/Library/LaunchAgents/jellyfinrpc.local.plist
```

Comment:
```
Set Jellyfin-RPC to run at login? (y/n) y
Unload failed: 5: Input/output error
Try running `launchctl bootout` as root for richer errors.
If needed, you can run Jellyfin RPC at any time by running 'jellyfin-rpc' in a terminal
```
I seem to be getting this error when selecting autostart

Reply:
[45582b4](https://github.com/Radiicall/jellyfin-rpc/pull/18/commits/45582b4ffe76510eb87e054575cba9818883b926#diff-3807b76448243f6b4d1b13c1f1b0f55ac4a802675378d7fd3fc76ba6e9a2e64cR163-R165) should fix this as well. Realized that I wasn't properly removing the LaunchAgent in my initial testing, so I'd incorrectly blamed the OS for it already being present because of the notification that appears when the LaunchAgent file is created. I assumed it needed unloaded before it could be manually loaded to run the binary.

Now the script properly checks for the existence of a LaunchAgent before attempting to remove it, in case a previous installation exists. But according to my testing, during a fresh install it will now cleanly load the new LaunchAgent.